MailerQ/Core/QueueManager.cs

```diff
@@ -72,7 +72,7 @@ public async Task PublishAsync(OutgoingMessage outgoingMessage, string queueName
             {
                 string jsonMessage = Encoding.UTF8.GetString(body);
                 var resultMessage = JsonConvert.DeserializeObject<T>(jsonMessage);
-                action.Invoke(resultMessage);
+                action.Invoke(resultMessage).Wait();
```

Comment:
![image](https://user-images.githubusercontent.com/1157864/65765870-c0f3db80-e0ff-11e9-980d-594828351df3.png)

Mmm, it is weird, maybe you are creating a task to run synchronously. I am not sure, but maybe you can make it purely async in order to take advantage of thread reuse.

Something like:

```csharp
bus.Consume(queue, (body, properties, info) => async () =>
{
    var jsonMessage = Encoding.UTF8.GetString(body);
    var resultMessage = JsonConvert.DeserializeObject<T>(jsonMessage);
    await action.Invoke(resultMessage);
```